elide-async/src/main/java/com/yahoo/elide/async/hooks/UpdateQueryPrincipalNameHook.java

```diff
@@ -11,17 +11,13 @@
 import com.yahoo.elide.security.ChangeSpec;
 import com.yahoo.elide.security.RequestScope;
 
-import java.security.Principal;
 import java.util.Optional;
 
-public class UpdatePrincipalNameHook implements LifeCycleHook<AsyncQuery> {
+public class UpdateQueryPrincipalNameHook extends AsyncHook implements LifeCycleHook<AsyncQuery> {
```

Comment:
I wonder if we should consolidate these three hooks into a single class.  A single hook can be called with different operation and phase arguments.  It might be more readable to have all the hook logic in one place.

Reply:
We don't have the phase available during execute to allow consolidation.